Provide System.CodeDom package readme (#107372)
dotnet/runtime: src/libraries/System.CodeDom/src/PACKAGE.md

## About

<!-- A description of the package and where one can find more documentation -->

Provides functionality for dynamically generating and compiling source code using the Code Document Object Model (CodeDOM).

It allows developers to represent code in a language-agnostic format and then generate code in multiple languages, such as C# and VB.NET.
The primary use cases include creating dynamic code generation tools, runtime code generation, and facilitating code analysis or transformation.

> 

Reviewer: Maybe add about Roslyn here? Something like:  
```suggestion

For a new modern development consider using Roslyn SDK, check [Cross-Platform Code Generation with Roslyn and .NET Core](https://learn.microsoft.com/archive/msdn-magazine/2017/may/net-core-cross-platform-code-generation-with-roslyn-and-net-core) and [Get started with source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators) for more info.

```

## About

<!-- A description of the package and where one can find more documentation -->

Provides functionality for dynamically generating and compiling source code using the Code Document Object Model (CodeDOM).

It allows developers to represent code in a language-agnostic format and then generate code in multiple languages, such as C# and VB.NET.
The primary use cases include creating dynamic code generation tools, runtime code generation, and facilitating code analysis or transformation.

> For a new modern development consider using the [.NET Compiler Platform SDK](https://learn.microsoft.com/otnet/csharp/roslyn-sdk/), check [Cross-Platform Code Generation with Roslyn and .NET Core](https://learn.microsoft.com/archive/msdn-magazine/2017/may/net-core-cross-platform-code-generation-with-roslyn-and-net-core) and [Get started with source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators) for more info.

Reviewer: NIT: typo, and removed the doc link that probably outdated, 
```suggestion
For a new modern development consider using the [.NET Compiler Platform SDK](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/), in particular [Roslyn source generators](https://learn.microsoft.com/dotnet/csharp/roslyn-sdk/source-generators-overview#get-started-with-source-generators).
```
